src/js_tests/wirecloud/ConnectionSpec.js

```diff
@@ -0,0 +1,251 @@
+ */
+
+/* jshint jasmine:true */
+/* globals Wirecloud */
+
+
+(function (ns) {
+
+    "use strict";
+
+    describe("Connection", function () {
+        var operatorModel, widgetModel, endpointDesc1, endpointDesc2, sourceEndpoint, targetEndpoint, wiringEngine;
+
+        beforeEach(function () {
+
+            endpointDesc1 = {
+                name: "source",
+                description: "description",
+                label: "title",
+                friendcode: "a"
+            };
+
+            endpointDesc2 = {
+                name: "target",
+                description: "description",
+                label: "title",
+                friendcode: "a"
+            };
+
+            operatorModel = {
+                id: "1",
+                volatile: false,
+                meta: {
+                    type: 'operator'
+                }
+            };
+
+            widgetModel = {
+                id: "1",
+                volatile: false,
+                meta: {
+                    type: 'widget'
+                }
+            };
+
+            wiringEngine = {
+                logManager: new Wirecloud.LogManager()
+            };
+
+            sourceEndpoint = new ns.WidgetSourceEndpoint(widgetModel, endpointDesc1);
+            targetEndpoint = new ns.OperatorTargetEndpoint(operatorModel, endpointDesc2);
+        });
+
+        describe("new Connection(wiringEngine, sourceEndpoint, targetEndpoint, [options])", function () {
+
+            it("should create a new instance with no options", function () {
+                var connection = new ns.Connection(wiringEngine, sourceEndpoint, targetEndpoint);
+
+                expect(connection instanceof ns.Connection).toBe(true);
+                expect(connection.source).toBe(sourceEndpoint);
+                expect(connection.target).toBe(targetEndpoint);
+                expect(connection.readonly).toBe(false);
+                expect(connection.volatile).toBe(false);
+                expect(connection.id).toEqual([sourceEndpoint.id, targetEndpoint.id].join("//"));
+            });
+
+            it("should create a new instance with options.readonly = true", function () {
+                var connection = new ns.Connection(wiringEngine, sourceEndpoint, targetEndpoint, {
+                    readonly: true
+                });
+
+                expect(connection.readonly).toBe(true);
+            });
+        });
+
+        describe("toJSON()", function () {
+
+            it("should parse to JSON", function () {
+                var connection = new ns.Connection(wiringEngine, sourceEndpoint, targetEndpoint);
+
+                expect(JSON.parse(JSON.stringify(connection))).toEqual({
+                    readonly: false,
+                    source: sourceEndpoint.toJSON(),
+                    target: targetEndpoint.toJSON()
+                });
+            });
+        });
+
+        describe("equals(value)", function () {
+
+            it("should be equal", function () {
```

Comment:
This method deserves more tests, for example, one for testing that the `equals` method returns `false` if the endpoints are different, another for testing it returns `true` if both connections are the same ....